petastorm/pytorch.py

```diff
@@ -348,7 +365,38 @@ def _instantiate_noop_buffer():
 
         return shuffling_buffer, other_shuffling_buffer, instantiate_buffer_fn
 
+    def _worker(self):
+        for b in self._iter_impl_worker():
+            if self._finished_event.is_set():
+                break
+            self._async_batch_queue.put(b)
+        else:
+            self._async_batch_queue.put(None)
+
     def _iter_impl(self):
+        if self._async_batch_queue:
+            # Get shuffled batches from async shuffle thread
+            if not self._started:
```

Comment:
If you think it's worth it to start the thread early (in the constructor), can we factor out this code so we don't have duplication?

Reply:
I see, I will think about it.